File: Android/iOrder/app/src/main/java/hr/fer/oobl/iorder/iorder/ui/login/LoginContract.java
package hr.fer.oobl.iorder.iorder.ui.login;

import android.os.Bundle;

import hr.fer.oobl.iorder.iorder.base.BaseView;
import hr.fer.oobl.iorder.iorder.base.ScopedPresenter;

public interface LoginContract {

    interface Presenter extends ScopedPresenter {

        void onInit(final Bundle savedInstanceState);

        void onLoginClicked(String email, String password);

        void onRegisterClicked();

        void onAnimationEnd();

        void cancel();
    }

    interface View extends BaseView {

        void displayLoginComponents();

        void animateLogo(final Bundle savedInstanceState);

        String getEmailInput();

        String getPasswordInput();

        void showProgress(boolean visible);

        void setButtons(boolean enable);

        void saveChanges();

        void hideErrors();

        void displayEmailInvalidError();

        void displayPasswordInvalidError();

        void processOfflineMode();
    }
}

File: Android/iOrder/app/src/main/java/hr/fer/oobl/iorder/iorder/ui/login/LoginContract.kt
package hr.fer.oobl.iorder.iorder.ui.login

import hr.fer.oobl.iorder.iorder.base.BaseView
import hr.fer.oobl.iorder.iorder.base.ScopedPresenter

/**
 * Created by mmihalic on 19.12.2017..
 */
interface LoginContract {

    interface View : BaseView {
        fun getEmailInput(): String

        fun getPasswordInput(): String

        fun showProgress(visible: Boolean)

        fun setButtons(enable: Boolean)

        fun saveChanges()
    }

    interface Presenter : ScopedPresenter
}